### DISCUSSION TITLE
Hide some Enum values from schema

### DISCUSSION THREAD

**Discussion by md84419** [fastapi/fastapi#9366] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum
from fastapi import Body, Query, FastAPI

class FormatModel(str,Enum):
    json5 = 'json5'
    csv = 'csv'
    json = 'json'

    @classmethod
    def choices(cls):
        return [x._value_ for x in cls if not hasattr(x, 'include_in_schema') or x.include_in_schema==True]

FormatModel.json.include_in_schema = False

app = FastAPI()

@app.api_route('/test', methods=['post'])
async def test(
    gloss=Body('test'),
    format: FormatModel = Query(
        ...,
        example=FormatModel.choices()[0],
        description=f"The format to return ({'|'.join(FormatModel.choices())})",
        required=True,
    ),
):
    pass
```

### Description

I would like to define a enum with some values to be hidden from the schema.
I would like to define my Query to use the enum.
I would like the API to work with all the values defined in the enum, but the docs, redoc and schema to only show a subset of the values.

Is there a way to do this in FastAPI?

#### Edit 2023-04-07T21:40Z+1:
I've found a workaround.

1. change the type of 'format' from FormatModel to string, and
2. use the undocumented enum argument to add enumerated values to the schema

```python
format: str = Query(
        ...,
        example=FormatModel.choices()[0],
        description=f"The format to return ({'|'.join(FormatModel.choices())})",
        required=True,
        enum=FormatModel.choices(),
    ),
```
This has the undesired side-effect that the API no longer validates that the supplied value is actually a member of FormatModel.
If one only does step 2, the schema ends up with an enumerated list consisting of both FormatModel and FormatModel.choices() (that is, with 5 members and with duplicates of 'json5' and 'csv').

### Operating System

Windows

### FastAPI Version

0.95

### Python Version

3.7

**Comment by tahayk** [comment]
Hey @md84419 ; 
probably you can use `underscore_attrs_are_private` in the [model config](https://docs.pydantic.dev/usage/model_config/) ?
